### ISSUE TITLE
Rails 6.0.3.4 joined where condition using enums no longer works

### ISSUE THREAD

**Issue by taurauko** [rails/rails#40758]
### Steps to reproduce

Have models in a configuration like:
```
Event
  has_one :schedule
  has_many :targets
Schedule
  belongs_to :event
  enum status: { queued: 0, started: 1, success: 2, failed: 3 }
Target
  belongs_to :event
```
Then try to fire a query like:
```
Target.joins(event: :schedule)
      .where(schedules: { status: 'queued' })
```

### Expected behavior

Work the same as Rails 5:
```
SELECT `targets`.*
FROM `targets`
INNER JOIN `events` ON `events`.`id` = `targets`.`event_id`
INNER JOIN `schedules` ON `schedules`.`event_id` = `events`.`id`
WHERE `schedules`.`status` = 0
```
### Actual behavior

The `status` is no longer replaced correctly with its integer value, instead silently using NULL:
```
SELECT `targets`.*
FROM `targets`
INNER JOIN `events` ON `events`.`id` = `targets`.`event_id`
INNER JOIN `schedules` ON `schedules`.`event_id` = `events`.`id`
WHERE `schedules`.`status` = NULL
```
At the same time, using the string value in a query still works in certain situations:
```
[1] pry(main)> Schedule.where(status: 'queued')
=>   Schedule Load (0.5ms)  SELECT `schedules`.* FROM `schedules` WHERE `schedules`.`status` = 0
```
It behaves the same using `:queued` instead of `'queued'`.

### System configuration
**Rails version**: 6.0.3.4

**Ruby version**: 2.6.5

**Comment by uxxman** [comment]
Test script

```ruby
# frozen_string_literal: true

require "bundler/inline"

gemfile(true) do
  source "https://rubygems.org"

  git_source(:github) { |repo| "https://github.com/#{repo}.git" }

  # Activate the gem you are reporting the issue against.
  gem "rails" # , github: "rails/rails"
  gem "sqlite3"
end

require "active_record"
require "minitest/autorun"
require "logger"

# This connection will do for database-independent bug reports.
ActiveRecord::Base.establish_connection(adapter: "sqlite3", database: ":memory:")
ActiveRecord::Base.logger = Logger.new(STDOUT)

ActiveRecord::Schema.define do
  create_table :events, force: true do |t|
  end

  create_table :schedules, force: true do |t|
    t.integer :event_id
    t.integer :status
  end

  create_table :targets, force: true do |t|
    t.integer :event_id
  end
end

class Event < ActiveRecord::Base
  has_one :schedule
  has_many :targets
end

class Schedule < ActiveRecord::Base
  belongs_to :event
  enum status: { queued: 0, started: 1, success: 2, failed: 3 }
end

class Target < ActiveRecord::Base
  belongs_to :event
end

class BugTest < Minitest::Test
  def test_association_stuff
    event = Event.create!
    Schedule.create! status: :queued, event: event
    Target.create! event: event

    assert_equal 1, Target.joins(event: :schedule).where(schedules: { status: 0 }).count
    assert_equal 1, Target.joins(event: :schedule).where(schedules: { status: :queued }).count
  end
end
```

Works as expected for rails 'master'.

**Comment by taurauko** [comment]
Indeed it works with rails master, but not the released 6.0.3.4

**Comment by ghiculescu** [comment]
This is fixed in https://github.com/rails/rails/pull/39465, but according to @kamipo it's not possible to backport (see https://github.com/rails/rails/issues/39460#issuecomment-635776991).
